File: ios/NSUserDefaultsSample/NSUserDefaultsSample/ViewController.m
//
//  ViewController.m
//  NSUserDefaultsSample
//
//  Created by fujita taisuke on 2012/07/29.
//  Copyright (c) 2012年 Revolution 9. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)viewDidUnload
{
    [super viewDidUnload];
    // Release any retained subviews of the main view.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {
        return (interfaceOrientation != UIInterfaceOrientationPortraitUpsideDown);
    } else {
        return YES;
    }
}

- (IBAction)respondToSaveButtonClick:(id)sender
{
    /*
     Person *tYamada = [[[Person alloc] init] autorelease];
     tYamada.name = @"山田太郎";
     Address *yAddress = [[[Address alloc] init] autorelease];
     yAddress.zipCode = @"104-0061";
     yAddress.state = @"東京都";
     yAddress.city = @"中央区";
     yAddress.other = @"銀座1丁目";
     tYamada.address = yAddress;
     
     Person *hYamada = [[[Person alloc] init] autorelease];
     hYamada.name = @"山田花子";
     hYamada.address = yAddress;
     
     Person *tanaka = [[[Person alloc] init] autorelease];
     tanaka.name = @"田中次郎";
     Address *tAddress = [[[Address alloc] init] autorelease];
     tAddress.zipCode = @"145-0071";
     tAddress.state = @"東京都";
     tAddress.city = @"大田区";
     tAddress.other = @"田園調布1丁目";
     tanaka.address = tAddress;
     NSData *data = [NSKeyedArchiver archivedDataWithRootObject:tYamada];
     NSData *data2 = [NSKeyedArchiver archivedDataWithRootObject:hYamada];
     NSData *data3 = [NSKeyedArchiver archivedDataWithRootObject:tanaka];
     NSArray *array = @[data, data2, data3];
     */
    // 初期値の設定
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    NSDictionary *dict = @{
    @"home" : @"http://www.google.co.jp",
    @"bookmarks": @[@"http://www.apple.com", @"http://www.9revolution9.com", @"https://twitter.com/"],
    @"font-size" : @14,
    @"javascript-enabled" : @(YES)
    };
    [defaults registerDefaults:dict];
    NSLog(@"%@", [defaults objectForKey:@"home"]);
    NSLog(@"%@", [defaults objectForKey:@"bookmarks"]);
    NSLog(@"%d", [defaults integerForKey:@"font-size"]);
    NSLog(@"%d", [defaults boolForKey:@"javascript-enabled"]);
    // データの保存
    NSArray *array = @[@"http://www.apple.com", @"http://www.9revolution9.com", @"https://twitter.com/"];
    [defaults setObject:array forKey:@"bookmarks"];
    BOOL successful = [defaults synchronize];
    if (successful) {
        NSLog(@"%@", @"データの保存に成功しました。");
    }
}

- (IBAction)respondToLoadButtonClick:(id)sender
{
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    NSArray *array = [defaults arrayForKey:@"bookmarks"];
    if (array) {
        for (NSString *data in array) {
            NSLog(@"%@", data);
        }
    } else {
        NSLog(@"%@", @"データが存在しません。");
    }
    NSLog(@"%@", [defaults objectForKey:@"home"]);
}

- (IBAction)respondToRemoveButtonClick:(id)sender
{
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    [defaults removeObjectForKey:@"bookmarks"];
    BOOL successful = [defaults synchronize];
    if (successful) {
        NSLog(@"%@", @"データの削除に成功しました。");
    } else {
        NSLog(@"%@", @"削除するデータがありません。");
        return;
    }
    // データが削除されていることを確認する
    NSArray *array = [defaults arrayForKey:@"bookmarks"];
    NSLog(@"%d:%@", successful, array);
    if (!array) {
        NSLog(@"%@", @"データは削除されました。");
    }
}

@end

File: ios_swift/UserDefaults2/UserDefaults2/ViewController.swift
//
//  ViewController.swift
//  UserDefaults2
//
//  Created by taisuke fujita on 2014/08/05.
//  Copyright (c) 2014年 taisuke fujita. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func respondToSaveButtonClick() {
        let defaults = NSUserDefaults.standardUserDefaults()
        // 山田太郎、花子オブジェクト(住所が同じ夫婦という設定)
        let address1 = Address(zipCode: "104-0061",
            state: "東京都", city: "中央区", other: "銀座1丁目")
        let taroYamada = Person(name: "山田太郎", address: address1)
        let hanakoYamada = Person(name: "山田花子", address: address1)
        // 田中次郎オブジェクト
        let address2 = Address(zipCode: "604-8126",
            state: "京都府", city: "京都市", other: "中京区")
        let jiroTanaka = Person(name: "田中次郎", address: address2)
        let archivedTaroYamada = NSKeyedArchiver.archivedDataWithRootObject(taroYamada)
        let archivedHanakoYamada = NSKeyedArchiver.archivedDataWithRootObject(hanakoYamada)
        let archivedJiroTanaka = NSKeyedArchiver.archivedDataWithRootObject(jiroTanaka)
        // 保存するデータを配列にまとめる
        let array = [archivedTaroYamada, archivedHanakoYamada, archivedJiroTanaka]
        defaults.setObject(array, forKey: "address-list")
        let successful = defaults.synchronize()
        if successful {
            println("データの保存に成功しました。")
        }
    }
    
    @IBAction func respondToLoadButtonClick() {
        let defaults = NSUserDefaults.standardUserDefaults()
        let addressList = defaults.arrayForKey("address-list") as [NSData]
        for data in addressList {
            let person = NSKeyedUnarchiver.unarchiveObjectWithData(data as NSData) as Person
            println(person.name)
            println(person.address.zipCode)
            println(person.address.state)
            println(person.address.city)
            println(person.address.other)
        }
    }
}

